drivers/bluetooth/hci/hci_psoc6_bless.c

```diff
@@ -0,0 +1,268 @@
+	.stackParam = &psoc6_bless_stack_param,
+	.hw = &psoc6_bless_hw_config,
+};
+
+static void psoc6_bless_rx_thread(void *, void *, void *)
+{
+	while (true) {
+		k_sem_take(&psoc6_bless_rx_sem, K_MSEC(BLE_THREAD_SEM_TMOUT_MS));
+		Cy_BLE_ProcessEvents();
+	}
+}
+
+static void psoc6_bless_isr_handler(const struct device *dev)
+{
+	if (Cy_BLE_HAL_BlessInterruptHandler()) {
+		k_sem_give(&psoc6_bless_rx_sem);
+	}
+}
+
+static void psoc6_bless_events_handler(uint32_t eventCode, void *eventParam)
+{
+	cy_stc_ble_hci_tx_packet_info_t *hci_rx = NULL;
+	struct net_buf *buf = NULL;
+	size_t buf_tailroom = 0;
+
+	if (wait_addr_set) {
+		wait_addr_set = false;
+		return;
+	}
+
+	if (eventCode != CY_BLE_EVT_HCI_PKT_RCVD) {
+		LOG_DBG("Other EVENT 0x%X", eventCode);
+		return;
+	}
+
+	hci_rx = eventParam;
+
+	switch (hci_rx->packetType) {
+	case PACKET_TYPE_HCI_EVENT:
+		buf = bt_buf_get_evt(hci_rx->data[0], 0, K_NO_WAIT);
+		if (!buf) {
+			LOG_ERR("Failed to allocate the buffer for RX: EVENT ");
+			return;
+		}
+
+		break;
+	case PACKET_TYPE_HCI_ACL_DATA:
+		buf = bt_buf_get_rx(BT_BUF_ACL_IN, K_NO_WAIT);
+		if (!buf) {
+			LOG_ERR("Failed to allocate the buffer for RX: ACL ");
+			return;
+		}
+		bt_buf_set_type(buf, BT_BUF_ACL_IN);
+
+		break;
+
+	default:
+		LOG_WRN("Unsupported HCI Packet Received");
+		return;
+	}
+
+	buf_tailroom = net_buf_tailroom(buf);
+	if (buf_tailroom < hci_rx->dataLength) {
+		LOG_WRN("Not enough space for rx data");
+		return;
+	}
+	net_buf_add_mem(buf, hci_rx->data, hci_rx->dataLength);
+	bt_recv(buf);
+}
+
+static int psoc6_bless_open(void)
+{
+	k_tid_t tid;
+
+	tid = k_thread_create(&psoc6_bless_rx_thread_data, psoc6_bless_rx_thread_stack,
+			      K_KERNEL_STACK_SIZEOF(psoc6_bless_rx_thread_stack),
+			      psoc6_bless_rx_thread, NULL, NULL, NULL,
+			      K_PRIO_COOP(CONFIG_BT_RX_PRIO), 0, K_NO_WAIT);
+	k_thread_name_set(tid, "psoc6_bless_rx_thread");
+
+	return 0;
+}
+
+static int psoc6_bless_send(struct net_buf *buf)
+{
+	cy_en_ble_api_result_t result;
+
+	memset((void *)&hci_tx_pkt, 0, sizeof(cy_stc_ble_hci_tx_packet_info_t));
+
+	hci_tx_pkt.dataLength = buf->len;
+	hci_tx_pkt.data = buf->data;
+
+	switch (bt_buf_get_type(buf)) {
+	case BT_BUF_ACL_OUT:
+		hci_tx_pkt.packetType = PACKET_TYPE_HCI_ACL_DATA;
+		break;
+	case BT_BUF_CMD:
+		hci_tx_pkt.packetType = PACKET_TYPE_HCI_COMMAND;
+		break;
+
+	default:
+		net_buf_unref(buf);
+		return -ENOTSUP;
+	}
+
+	if (k_sem_take(&psoc6_bless_operation_sem, K_MSEC(BLE_LOCK_TMOUT_MS)) != 0) {
+		LOG_ERR("Failed to acquire BLE DRV Semaphore");
+		net_buf_unref(buf);
+		return -EIO;
+	}
+
+	result = Cy_BLE_SoftHciSendAppPkt(&hci_tx_pkt);
+	if (result != CY_BLE_SUCCESS) {
+		LOG_ERR("Error in sending packet reason %d\r\n", result);
+	}
+
+	k_sem_give(&psoc6_bless_operation_sem);
+
+	net_buf_unref(buf);
+
+	return 0;
+}
+
+static int psoc6_bless_setup(void)
+{
+#define BT_HCI_VND_OP_SET_PUBLIC_ADDR     0xFDA0
+#define BD_ADDR_TYPE_PUBLIC               0
```

Comment:
These are already defined in the public hci.h header (BT_LE_ADDR_TYPE_ PUBLIC etc)

Reply:
Correction, they’re in addr.h